logging/src/main/java/life/qbic/logging/impl/publisher/SimplePublisher.java

```diff
@@ -0,0 +1,119 @@
+package life.qbic.logging.impl.publisher;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Queue;
+import java.util.concurrent.ArrayBlockingQueue;
+import life.qbic.logging.api.LogMessage;
+import life.qbic.logging.api.Publisher;
+import life.qbic.logging.api.Subscriber;
+
+/**
+ * Basic implementation of the {@link Publisher} interface.
+ *
+ * @since 1.0.0
+ */
+class SimplePublisher implements Publisher {
+
+  private static final int DEFAULT_MESSAGE_CAPACITY = 100;
+
+  private final Collection<Subscriber> subscribers;
+
+  private final Queue<LogMessage> logMessages;
+
+  public SimplePublisher() {
+    subscribers = Collections.synchronizedCollection(new HashSet<>());
+    logMessages = new ArrayBlockingQueue<>(DEFAULT_MESSAGE_CAPACITY);
+    Broadcasting broadcasting = new Broadcasting(subscribers, this);
+    broadcasting.start();
+  }
+
+  @Override
+  public void subscribe(Subscriber s) {
+    requireNonNull(s, "Subscriber must not be null");
+    synchronized (subscribers) {
+      subscribers.add(s);
+    }
+  }
+
+  @Override
+  public void unsubscribe(Subscriber s) {
+    requireNonNull(s, "Subscriber must not be null");
+    synchronized (subscribers) {
+      subscribers.remove(s);
+    }
+  }
+
+  @Override
+  public synchronized void publish(LogMessage logMessage) {
+    requireNonNull(logMessage, "LogMessage must not be null");
+    logMessages.add(logMessage);
+    notifyAll();
+  }
+
+  protected synchronized LogMessage nextLogMessage() {
+    while (logMessages.isEmpty()) {
+      try {
+        wait();
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+      }
+    }
+    return logMessages.remove();
+  }
```

Comment:
As you modify the `logMessages` queue here, do you need to notify any other thread?

Reply:
I think this is implemented in the ArrayBlockingQueue class of the Java collections package

Reply:
reading the API i switch it to `poll` instead of `remove`, because i dont want an ElementNotFoundException thrown here, but return null if the queue is empty.

Reply:
I could not infer that from the documentation.
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/ArrayBlockingQueue.html
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/AbstractQueue.html#remove()